packages/block-library/src/navigation/interactivity.js

```diff
@@ -45,62 +73,51 @@ store( {
 	selectors: {
 		core: {
 			navigation: {
-				roleAttribute: ( { context } ) => {
-					return context.core.navigation.overlay &&
-						context.core.navigation.isMenuOpen
+				roleAttribute: ( { context } ) =>
+					context.core.navigation.overlay &&
+					context.core.navigation.isMenuOpen
```

Comment:
Should we use here `selectors.core.navigation.isMenuOpen( { context } )`? Right now, the role attribute is always `dialog` because that context is always truthy (I believe).

Reply:
Oh, my bad. I missed that, thanks. I'll fix it now.

Reply:
Is this enough? I'm still learning these accessibility quirks, so I'm not sure when `roleAttribute` should change.

https://github.com/WordPress/gutenberg/pull/50953/commits/07b41f91b3381a4e61c5f92556c3d48abf8004b8

Reply:
Yes, that should be enough 🙂 At least, that is how it is working with the current implementation, I believe.

Reply:
Nice, thanks!